library/src/main/java/com/novoda/downloadmanager/lib/Request.java

```diff
@@ -246,6 +247,15 @@ public Request allowTarUpdates() {
     }
 
     /**
+     * When a ETag header is present, the application should check the integrity of the
+     * downloaded file, otherwise the current download won't be able to be resumed
```

Comment:
I didn't know about that check as well so this is what I found out:

```
/**
         * The name of the column containing the flags that indicates whether
         * the initiating application is capable of verifying the integrity of
         * the downloaded file. When this flag is set, the download manager
         * performs downloads and reports success even in some situations where
         * it can't guarantee that the download has completed (e.g. when doing
         * a byte-range request without an ETag, or when it can't determine
         * whether a download fully completed).
         * <P>Type: BOOLEAN</P>
         * <P>Owner can Init</P>
         */
        public static final String COLUMN_NO_INTEGRITY = "no_integrity";
```

And the Google coded this:
```
public boolean isResumable() {
        return alwaysResume || (eTag != null && isNoIntegrity());
    }
```

So I assumed that they forced that if there is an eTag, the flag `noIntegrity` is checked, leading at that if the default flag is false, with an eTag present, the download is non resumable